src/azure-cli/azure/cli/command_modules/appconfig/feature.py

```diff
@@ -633,6 +633,136 @@ def add_filter(cmd,
         "Failed to add filter for the feature flag '{}' due to a conflicting operation.".format(feature))
 
 
+def update_filter(cmd,
+                  filter_name,
+                  feature=None,
+                  key=None,
+                  name=None,
+                  label=None,
+                  filter_parameters=None,
+                  yes=False,
+                  index=None,
+                  connection_string=None,
+                  auth_mode="key",
+                  endpoint=None):
+    if key is None and feature is None:
+        raise RequiredArgumentMissingError(
+            "Please provide either `--key` or `--feature` value.")
+    if key and feature:
+        logger.warning(
+            "Since both `--key` and `--feature` are provided, `--feature` argument will be ignored.")
+
+    key = FeatureFlagConstants.FEATURE_FLAG_PREFIX + feature if key is None else key
+    # Get feature name from key for logging. If users have provided a different feature name, we ignore it anyway.
+    feature = key[len(FeatureFlagConstants.FEATURE_FLAG_PREFIX):]
+
+    azconfig_client = get_appconfig_data_client(
+        cmd, name, connection_string, auth_mode, endpoint)
+
+    if index is None:
+        index = float("-inf")
+
+    # Construct feature filter
+    if filter_parameters is None:
+        filter_parameters = {}
+    new_filter = FeatureFilter(filter_name, filter_parameters)
+
+    retry_times = 3
+    retry_interval = 1
+    for i in range(0, retry_times):
+        try:
+            retrieved_kv = azconfig_client.get_configuration_setting(
+                key=key, label=label)
+        except ResourceNotFoundError:
+            raise CLIError(
+                "Feature flag '{}' with label '{}' not found.".format(feature, label))
+        except HttpResponseError as exception:
+            raise CLIError(
+                "Failed to retrieve feature flags from config store. " + str(exception))
```

Comment:
Could you raise specific error type from `azure.cli.core.azclierror` instead of `CLIError`?

Reply:
Since this is the case also with other errors raised across other methods in the app config module, would it be possible to create a separate PR for this which will raise specific errors for all other methods? That way, it could also be reviewed by a member of the team and this current PR would not be delayed.

Reply:
OK, sounds good~

Reply:
Thanks!